setup.py

```diff
@@ -37,18 +37,18 @@
     # Allow installation on 2.7.9+, and 3.4+ even though we officially only support 3.6+.
     python_requires=">=2.7.9, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
     install_requires = [
-        'intelhex',
-        'six',
-        'enum34;python_version<"3.4"',
+        'intelhex>=2.0,<3.0',
+        'six>=1.0,<2.0',
+        'enum34>=1.0,<2.0;python_version<"3.4"',
         'future',
         'websocket-client',
-        'intervaltree<3.0',
+        'intervaltree>=2.0,<3.0',
```

Comment:
We should be able to set intervaltree to >=3.0.1,<4.0. The problem that triggered all this was fixed, and the 3.x versions fix a number of warnings caused by use of a deprecated API.

Reply:
I bumped it to 3.0.2 since it looks like there was an [encoding fix](https://github.com/chaimleib/intervaltree/releases/tag/3.0.2):

"On some systems, setup.py opened README.md with a non-unicode encoding. My fault for leaving the encoding flapping in the breeze. It's been fixed."